cli/version/version_match.go

```diff
@@ -0,0 +1,161 @@
+package version
+
+import (
+	"fmt"
+
+	"github.com/tarantool/tt/cli/util"
+)
+
+type requiredField uint8
+type requiredFields []requiredField
+
+const (
+	// Values according order of Fields in regex from `matchVersionParts()`.
+	requiredFieldBuildName requiredField = iota
+	requiredFieldMajor
+	requiredFieldMinor
+	requiredFieldPatch
+	requiredFieldReleaseType
+	requiredFieldReleaseNum
+	requiredFieldAdditional
+	requiredFieldHash
+	requiredFieldRevision
+
+	countOfRequiredFields
+)
+
+const NotFoundVersion = "NOT_FOUND"
+
+// exploreMatchVersion examines the given string and retrieves the version components,
+// that define the rules for checking for version consistency.
+func exploreMatchVersion(verStr string) (Version, requiredFields, error) {
+	var version Version
+	fields := make(requiredFields, 0, countOfRequiredFields)
+
+	matches, err := matchVersionParts(verStr, false)
```

Comment:
Now I can install version with incorrect name. For example if I try to use versions like `v.2.1`, `.2.1` the output will be:

```bash
$ ./tt install tt v.2.1 
   • Searching in versions... 
   • Checking existing...     
   • Installing tt=v2.2.1     
   • Checking dependencies... 
   • Downloading tt...        
   • Building tt...           
   • Copying executable...    
   • Tt executable is installed to: "/home/milchenko/tarantool/tt/bin/tt_v2.2.1"
   • Made default by symlink "/home/milchenko/tarantool/tt/bin/tt"
   • Use the following command to add the bin_dir directory to the PATH: . <(tt env)
   • Done.    
```

Possibly it can be a feature due to `tt` try to search the latest version by giving pattern even it has incorrect syntax, but in my opinion it is more a bug. Correct me if I'm wrong.

Reply:
It seems debatable - what the correct version means.
Here you apply the rule where <major> does not important. And it found first of last version with such match.

If needed, we can set the <major> part as mandatory. But, the best way here should be more interactive behavior of the tool. It should collect all matches and ask user to make a choose.

Reply:
To be honest, I don't know when such behavior might be useful. So I propose require a major if minor is set, minor if patch is set etc.

Reply:
Implement this logic:
- `patch` - requires `minor`
- `minor` - requires `major`